src/univariate/continuous/burr.jl

```diff
@@ -0,0 +1,194 @@
+"""
+    Burr(k, c, λ)
+The *Burr distribution*  with shape1 `k`, shape2 `c` and scale `λ` has probability density function
+```math
+f(x; k, c, \\lambda) = \\frac{ck}{\\lambda} \\left(\\frac{x}{\\lambda}\\right)^{c-1} \\left[ 1+\\left(\\frac{x}{\\lambda}\\right)^{c} \\right]^{-k-1}, \\quad x > 0
+```
+```julia
+Burr()                  # Burr distribution with k = 1, c = 1 and λ = 1, i.e. Burr(1, 1, 1)
+Burr(k, c)              # Burr distribution with λ = 1, i.e. Burr(k, c, 1)
+Burr(k, c, λ)           # Burr distribution with shape1 k, shape2 c and scale λ
+params(d)        # Get the parameters, i.e. (k, c, λ)
+```
+External links
+* [Burr distribution on Wikipedia](https://en.wikipedia.org/wiki/Burr_distribution)
+"""
+
+
+struct Burr{T<:Real} <: ContinuousUnivariateDistribution
+    k::T # shape1
+    c::T # shape2
+    λ::T # scale
+end
+
+function Burr(k::T, c::T, λ::T; check_args=true) where {T}
+    check_args && @check_args(Burr, k > zero(k) && c > zero(c) && λ > zero(λ))
+    return Burr{T}(k, c, λ)
+end
+
+Burr(k::Real, c::Real, λ::Real) = Burr(promote(k, c, λ)...)
+Burr(k::Integer, c::Integer, λ::Integer) = Burr(float(k), float(c), float(λ))
+Burr(k::T, c::T) where {T <: Real} = Burr(k, c, one(T))
+Burr() = Burr(1.0, 1.0, 1.0, check_args=false)
+
+@distr_support Burr 0.0 Inf
+
+#### Conversions
+
+function convert(::Type{Burr{T}}, k::S, c::S, λ::S) where {T <: Real, S <: Real}
+    Burr(T(k), T(c), T(λ))
+end
+function convert(::Type{Burr{T}}, d::Burr{S}) where {T <: Real, S <: Real}
+    Burr(T(d.k), T(d.c), T(d.λ), check_args=false)
+end
+
+#### Parameters
+
+# shape1 and shape2 not defined in the package -> ignore these. params is sufficient
+# shape1(d::Burr) = d.k
+# shape2(d::Burr) = d.c
+# scale(d::Burr) = d.λ
+params(d::Burr) = (d.k, d.c, d.λ)
+partype(::Burr{T}) where {T<:Real} = T
+
+#### Statistics
+
+function m(d::Burr{T}, g::Real) where T<:Real # The calculating moments: this should not be exported.
+    (k, c, λ) = params(d)
+
+    if (g-(-c))*(g-k*c) < 0
+        λ^g * gamma(1 + g/c) * gamma(k - g/c) / gamma(k)
+    else
+        return T(Inf)
+    end
+
+end
+
+
+function median(d::Burr{T}) where T<:Real
+    (k, c, λ) = params(d)
+
+    return λ * (2^(1/k) - 1)^(1/c)
+end
+
+function mean(d::Burr{T}) where T<:Real
+    return m(d, 1)
+end
```

Comment:
```suggestion
mean(d::Burr) = m(d, 1)
```
As said above, a more descriptive name for `m` would be good.

Reply:
Changed as commented.